model/dyn_core.F90

```diff
@@ -1078,6 +1081,72 @@ subroutine dyn_core(npx, npy, npz, ng, sphum, nq, bdt, n_map, n_split, zvir, cp,
+          call start_group_halo_update(i_pack(8), u, v, domain, gridtype=DGRID_NE)
+          call start_group_halo_update(i_pack(10), q, domain)
+          call start_group_halo_update(i_pack(11), q_con, domain)
+          call start_group_halo_update(i_pack(12), cappa, domain)
+          call complete_group_halo_update(i_pack(1), domain)
+          call complete_group_halo_update(i_pack(7), domain)
+          call complete_group_halo_update(i_pack(8), domain)
+          call complete_group_halo_update(i_pack(10), domain)
+          call complete_group_halo_update(i_pack(11), domain)
+          call complete_group_halo_update(i_pack(12), domain)
+          call timing_off('COMM_TOTAL')
+       
+          call fast_phys (is, ie, js, je, isd, ied, jsd, jed, npz, npx, npy, nq, &
+             flagstruct%c2l_ord, dt, consv, akap, ptop, phis, te0_2d, u, v, w, pt, &
+             delp, delz, q_con, cappa, q, pkz, zvir, flagstruct%te_err, flagstruct%tw_err, &
+             gridstruct, domain, bd, hydrostatic, do_adiabatic_init, &
+             flagstruct%consv_checker, flagstruct%adj_mass_vmr)
+
+          call timing_on('COMM_TOTAL')
+          call start_group_halo_update(i_pack(1), delp, domain, complete=.false.)
+          call start_group_halo_update(i_pack(1), pt, domain, complete=.true.)
+          call start_group_halo_update(i_pack(7), w, domain)
+          call start_group_halo_update(i_pack(8), u, v, domain, gridtype=DGRID_NE)
+          call start_group_halo_update(i_pack(10), q, domain)
+          call start_group_halo_update(i_pack(11), q_con, domain)
+          call start_group_halo_update(i_pack(12), cappa, domain)
+          call complete_group_halo_update(i_pack(1), domain)
+          call complete_group_halo_update(i_pack(7), domain)
+          call complete_group_halo_update(i_pack(8), domain)
+          call complete_group_halo_update(i_pack(10), domain)
+          call complete_group_halo_update(i_pack(11), domain)
+          call complete_group_halo_update(i_pack(12), domain)
+          call timing_off('COMM_TOTAL')
+
+          if (remap_step) then
+              pe (is:ie, 1, js:je) = ptop
+              peln (is:ie, 1, js:je) = log (pe (is:ie, 1, js:je))
+              pk (is:ie, js:je, 1) = exp (akap * peln (is:ie, 1, js:je))
+              do k = 2, npz + 1
+                  do j = js, je
+                      do i = is, ie
+                          pe (i, k, j) = pe (i, k-1, j) + delp (i, j, k-1)
+                          peln (i, k, j) = log (pe (i, k, j))
+                          pk (i, j, k) = exp (akap * peln (i, k, j))
+                      enddo
+                  enddo
+              enddo
+              call pe_halo (is, ie, js, je, isd, ied, jsd, jed, npz, ptop, pe, delp)
+          endif
+
+      endif
+
+!-----------------------------------------------------------------------
+! <<< Fast Physics
+!-----------------------------------------------------------------------
```

Comment:
This whole block of start/completes should be encapsulated within the fast_phys routine.  Also, there is no need to use the start/complete here as there is no attempt to overlap communications and compute.  I'm sure the same could have been accomplished with simple mpp_update_domain calls using false/true completions to pack everything together.

Reply:
@linjiongzhou was this code that you developed?

Reply:
Yes, I wrote this code. I will rewrite it. If it causes trouble, we don't need to include it in this release.

Reply:
@lharris4 @bensonr would you like this rewrite done for this release? If we do include it now I would like some time to retest the release.

Reply:
It can be done in a future release.

Reply:
I will rewrite it in a future release.